src/utils/private/buffer_impl.cpp

```diff
@@ -270,7 +292,11 @@ class Buffer::BufferImpl {
 			throw_exception(EXC_INVALID_PARAMETER, "Device not output buffer capable, so no buffer was created");
 		}
 
-		destroy();
+		if (nb_samples != m_last_nb_samples || cyclic) {
```

Comment:
Can you explain this in a couple of words (some comments)? What does libiio do if we try to do this when in cyclic/non cyclic mode.

Reply:
Added some comments in commit 0377d99.